tests/tracing/test_instrument_tracing.py

```diff
@@ -146,11 +146,25 @@ def func_with_exception():
 def test_instrument_func_disabled(monkeypatch):
     monkeypatch.setenv("OTEL_TRACING", "false")
     # Recreate an instance of TracingWrapper
-    tw = TracingWrapper()
+    tw = ServerTracingWrapper()
 
     @tw.instrument_func()
     def foo():
         return 1
 
     assert foo() == 1
     assert tw.provider is None
+
+
+def test_client_tracing_fallback(monkeypatch):
+    def mock_import(name, *args, **kwargs):
```

Comment:
I don't get how this test could work:

`pubtools._impl.tracing` has already been imported and `OPENTELEMETRY_AVAILABLE` initialized at line 7 of this file. This is long before `test_client_tracing_fallback` has been invoked, and before `monkeypatch.setattr` is called.

Have you confirmed that this code is actually reached?